static/scripts/view.js

```diff
@@ -789,18 +789,160 @@ function checkFormValidity(event) {
   return valid;
 }
 
+
+var isWaitingOnAsyncCall = false;
+var hasShownSubmissionMessage = false;
+var isSubmittingChanges = false;
+
+function checkBugChange(bugid, lastMod, inMsg) {
+  var tiqitApi = new TiqitApi();
+  var msg = inMsg;
+
+  // Prepare the async event functions
+  bugChangeOnLoad = function(eventLoad) {
+    var submitChanges = true;
+
+    if (eventLoad.target.status == 200 && eventLoad.target.responseXML) {
+      // Check if the first date predates lastMod and only Note or Attachment
+      // related changes were made.
+      // If not, we may have overwriteable changes. (See hasHistoryOverwriteableChanges).
+      // If this is the case, check with the user if bug changes should be submitted.
+      var history = historyDataFromXML(eventLoad.target.responseXML);
+      var lastModHistoryDate = new Date(history[0]['Date']);
+      var lastModDate = new Date(lastMod);
+      var hasHistoryOverwriteableChanges = false;
+
+      // Checking for legitimate changes here
+      if (lastModHistoryDate > lastModDate) {
+        for (i in history) {
+          var historyItem = history[i];
```

Comment:
I don't think it'll change anything here, but worth pointing out an oddity of javascript if you aren't already familiar - "variable hoisting". I'm pretty sure that the variable `historyItem` declared here is actually the *same* variable as the one in the loop lower down (since they're in the same function scope). I think it's fine because you're always setting a value, but strange things can happen if you aren't careful!

Reply:
Wow, had no idea that was the case. I'll clean up the code to only have one declaration.